homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,263 @@
+        """Return whether the monitor has alerted."""
+        return self.device.alerted
+
+    @property
+    def is_detected(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.detected
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.device.client.is_available
+
+    @property
+    def connected(self) -> bool:
+        """Return True if entity is connected."""
+        return self.device.connected
+
+    @property
+    def enabled(self) -> bool:
+        """Return True if entity is enabled."""
+        return self.device.online
+
+    @property
+    def supported_features(self) -> int:
+        """Return supported features."""
+        return SUPPORT_ON_OFF | SUPPORT_STREAM
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if on."""
+        return self.device.online
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        if self.enabled:
+            return "mdi:camcorder"
+        return "mdi:camcorder-off"
+
+    @property
+    def state(self):
+        """Return the camera state."""
+        if self.device.recording:
+            return STATE_RECORDING
+        if self.device.online:
+            return STATE_STREAMING
+        return STATE_IDLE
+
+    async def stream_source(self) -> str:
+        """Return the mp4 stream source."""
+        return self._stream_url
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this agent object."""
+        return self._unique_id
+
+    async def async_enable_alerts(self):
+        """Enable alerts."""
+        await self.device.alerts_on()
+
+    async def async_disable_alerts(self):
+        """Disable alerts."""
+        await self.device.alerts_off()
+
+    async def async_enable_motion_detection(self):
+        """Enable motion detection."""
+        await self.device.detector_on()
+
+    async def async_disable_motion_detection(self):
+        """Disable motion detection."""
+        await self.device.detector_off()
+
+    async def async_start_recording(self):
+        """Start recording."""
+        await self.device.record()
+
+    async def async_stop_recording(self):
+        """Stop recording."""
+        await self.device.record_stop()
+
+    async def async_turn_on(self):
+        """Enable the camera."""
+        await self.device.enable()
+
+    async def async_toggle(self):
+        """Enable/disable the camera."""
+        if self.device.online:
+            await self.device.disable()
+        else:
+            await self.device.enable()
+
+    async def async_snapshot(self):
+        """Take a snapshot."""
+        await self.device.snapshot()
+
+    async def async_turn_off(self):
+        """Disable the camera."""
+        await self.device.disable()
+
+    # Other Entity method overrides
+
+    async def async_on_demand_update(self):
```

Comment:
Please make this a callback instead of a coroutine as we don't need to await inside. There's an overhead to awaiting a coroutine compared to calling a callback.

Decorate the method with `@callback` imported from core.py.

Reply:
so just add @callback to it? I can't find an example of anyone else doing this in any other integration

Reply:
code block has been removed